php/libraries/SinglePointLogin.class.inc

```diff
@@ -515,5 +516,42 @@ class SinglePointLogin
         $result =  $DB->insert('user_login_history', $setArray);
     }
 
+    /**
+     * Checks if a user has not logged in for a given amount of time
+     * As a side-effect, if true, will also set user field 'active' to 'N'
+     * Need to have already confirmed valid user
+     *
+     * @param String $username
+     * @param Int    $maximumDaysInactive Longest allowable time for inactivity in days
+     *
+     *
+     * @return bool - true if the user has now been disabled
+     */
+    function disabledDueToInactivity($username, $maximumDaysInactive)
+    {
+        $DB =& Database::singleton();
+        $query = "SELECT Login_timestamp 
+                  FROM (SELECT Login_timestamp, UserID 
+                        FROM user_login_history 
+                        WHERE UserID = :username 
+                        ORDER BY Login_timestamp DESC) 
+                        as t1 
+                  GROUP BY UserID";
```

Comment:
I'm a little confused by the reasoning for this query. It seems like you're just trying to do:

`SELECT MAX(Login_timestamp) FROM user_login_history WHERE UserID=:username`